incubator/hnc/internal/reconcilers/object.go

```diff
@@ -787,14 +790,29 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	}
 }
 
+// getSelector returns the selector on a given object if it exists
+func (r *ObjectReconciler) getTreeSelector(log logr.Logger, inst *unstructured.Unstructured) labels.Selector {
+	annot := inst.GetAnnotations()
+	treeSelectorStr, ok := annot[api.AnnotationTreeSelector]
+	if !ok {
+		return nil
+	}
+	return r.getSelectorFromString(log, treeSelectorStr+api.LabelTreeDepthSuffix)
```

Comment:
Isn't `LabelTreeDepthSuffix` already included in the `treeSelectorStr`?

Reply:
No, the `treeSelectorStr` is a short-handed representation of `selectorStr`, which doesn't include the `LabelTreeDepthSuffix`, and we need to add that in

Reply:
Ah right, sorry for the confusion. Selector is the one with DepthSurfix and treeSelector is the one only with namespace names.